File: SPCode/Modules/FormSupport.vb

Module FormSupport
    Public Sub FormCenter(ByRef frmCurrent As Form)
        Dim intLeft As Integer
        Dim intTop As Integer
        Dim frm As Form

        frm = CType(frmCurrent, Form)
        intLeft = (Screen.PrimaryScreen.Bounds.Width \ 2) - (frmCurrent.Width \ 2)
        intTop = (Screen.PrimaryScreen.Bounds.Height \ 2) - (frmCurrent.Height \ 2)
        With frm
            .Location = New Point(intLeft, intTop)
        End With
    End Sub
    Public Function FormClosingCheck(ByRef frmName As Form) As DialogResult
        If frmName.WindowState = System.Windows.Forms.FormWindowState.Minimized Then
            frmName.WindowState = FormWindowState.Normal
        End If

        frmName.BringToFront()

        ' Ask user what to do with the changes
        Return MessageBox.Show(TextDictionary("MS_DATACHANGED"), frmName.Text, MessageBoxButtons.YesNoCancel)
    End Function
    Public Sub FormCenterChild(ByRef frmCurrent As Form)
        Dim intLeft As Integer
        Dim intTop As Integer
        Dim frm As Form

        frm = CType(frmCurrent, Form)
        intLeft = (Common.MainShell.Width \ 2) - (frmCurrent.Width \ 2)
        intTop = (Common.MainShell.Height \ 2) - (frmCurrent.Height \ 2)
        With frm
            .Location = New Point(intLeft, intTop)
        End With
    End Sub
    Public Function DeleteAsk(ByVal strMsg As String, Optional ByRef strCaption As String = "") As Boolean
        If MsgBox(strMsg, MsgBoxStyle.Question Or MsgBoxStyle.YesNo, IIf(strCaption <> "", strCaption, System.Windows.Forms.Form.ActiveForm.Text)) = MsgBoxResult.Yes Then
            DeleteAsk = True
        Else
            DeleteAsk = False
        End If
    End Function
End Module

File: SPCode/CS/Modules/FormSupport.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;


namespace SoftLogik.Win
{
	sealed class FormSupport
	{
		public static void FormCenter(Form frmCurrent)
		{
			int intLeft;
			int intTop;
			Form frm;
			
			frm = frmCurrent;
			intLeft = (Screen.PrimaryScreen.Bounds.Width / 2) - (frmCurrent.Width / 2);
			intTop = (Screen.PrimaryScreen.Bounds.Height / 2) - (frmCurrent.Height / 2);
			frm.Location = new Point(intLeft, intTop);
		}
		public static DialogResult FormClosingCheck(Form frmName)
		{
			if (frmName.WindowState == System.Windows.Forms.FormWindowState.Minimized)
			{
				frmName.WindowState = FormWindowState.Normal;
			}
			
			frmName.BringToFront();
			
			// Ask user what to do with the changes
			return MessageBox.Show(CultureSupport.TextDictionary("MS_DATACHANGED", TextReturnTypeEnum.PureString), frmName.Text, MessageBoxButtons.YesNoCancel);
		}
		public static void FormCenterChild(Form frmCurrent)
		{
			int intLeft;
			int intTop;
			Form frm;
			
			frm = frmCurrent;
			intLeft = (Common.MainShell.Width / 2) - (frmCurrent.Width / 2);
			intTop = (Common.MainShell.Height / 2) - (frmCurrent.Height / 2);
			frm.Location = new Point(intLeft, intTop);
		}
		public static bool DeleteAsk(string strMsg, string strCaption)
		{
			bool returnValue;
			if (Interaction.MsgBox(strMsg, MsgBoxStyle.Question || MsgBoxStyle.YesNo, (strCaption != "" ? strCaption : System.Windows.Forms.Form.ActiveForm.Text)) == MsgBoxResult.Yes)
			{
				returnValue = true;
			}
			else
			{
				returnValue = false;
			}
			return returnValue;
		}
	}
	
	
}
